### PR TITLE
Swift: flow through keypath optional components

### PR DESCRIPTION
**Pull request by rdmarsh2**

### CODE REVIEW

File: swift/ql/lib/codeql/swift/dataflow/internal/DataFlowPrivate.qll [hunk 1]
```diff
@@ -727,6 +727,14 @@ predicate storeStep(Node node1, ContentSet c, Node node2) {
     c.isSingleton(any(Content::ArrayContent ac))
   )
   or
+  // Store throug
```
Comment (MathiasVP) [inline swift/ql/lib/codeql/swift/dataflow/internal/DataFlowPrivate.qll:730]: I think this sentence needs a bit of love 😂

**Review by MathiasVP — COMMENTED** [review]
LGTM otherwise

**Review by geoffw0 — APPROVED** [review]
LGTM too.  There are some analysis time regressions on the DCA run, though they look more like wobbles than anything related to this PR.

**Comment by MathiasVP** [comment]
> LGTM too. There are some analysis time regressions on the DCA run, though they look more like wobbles than anything related to this PR.

Indeed, there's a couple of stage-timing changes. But since nothing is reflected in the end-to-end it's probably just wobble.

**Pull request merged**